# Final Project Results Summary

This notebook summarizes the final performance metrics from both traditional ML models and Deep Reinforcement Learning agents.

## Traditional Models (GBM, DNN)
These models were evaluated on Mean Squared Error (MSE) and R².

In [1]:
import pandas as pd
traditional_results = {'GBM': {'MSE': 5.5986, 'R2': 0.99775}, 'DNN': {'MSE': 485.60193, 'R2': 0.8189}}
trad_df = pd.DataFrame(traditional_results).T
trad_df

MSE       R2
GBM    5.59860  0.99775
DNN  485.60193  0.81890

## DRL Agents (DQN, PPO, A2C)
These models were evaluated on total cumulative reward over the episode.

In [2]:
drl_results = {'DQN': {'Total Reward': 15.84}, 'PPO': {'Total Reward': 65.76}, 'A2C': {'Total Reward': 161.04}}
drl_df = pd.DataFrame(drl_results).T
drl_df

Total Reward
DQN         15.84
PPO         65.76
A2C        161.04


## 📌 Summary & Interpretation

- **GBM** achieved the highest prediction accuracy (R² ≈ 0.998), confirming its superiority for regression.
- **DNN** lagged behind GBM in accuracy, but still performed reasonably well.
- **A2C** delivered the highest cumulative reward among DRL models.
- **PPO** showed moderate performance, while **DQN** had the lowest total reward.

This concludes the project’s modeling and evaluation stage, consistent with the proposal’s goals.

